sw/device/lib/dif/dif_kmac.c

```diff
@@ -4,4 +4,55 @@
 
 #include "sw/device/lib/dif/dif_kmac.h"
 
+#include "sw/device/lib/base/memory.h"
+
 // TODO: implement!
+
+dif_kmac_result_t dif_kmac_customization_string_init(
+    const char *data, size_t len, dif_kmac_customization_string_t *out) {
+  if ((data == NULL && len != 0) || out == NULL) {
+    return kDifKmacBadArg;
+  }
+
+  if (len > kDifKmacMaxCustomizationStringLen) {
+    return kDifKmacBadArg;
+  }
+
+  // Convert length in bytes to length in bits.
+  uint16_t bits = (uint16_t)(len)*8;
+
+  // `left_encode` length in bits.
```

Comment:
Would it make sense to leave a note here explaining that the buffer is `kDifKmacMaxCustomizationStringOverhead + kDifKmacMaxCustomizationStringLen` long, so we can add the 3 bytes here without overflowing? I raised my eyebrows when reading the patch and had to convince myself that all was well...

Reply:
I was thinking a static assert here might provide some additional safety as well as documenting the size of the array relative to `kDifKmacMaxCustomizationStringLen`. Something like `_Static_assert(ARRAYSIZE(out->buffer) == kDifKmacMaxCustomizationStringLen + 2)`. What do you think?

Reply:
Sounds good to me! Maybe `>=`, because then you decouple `3` and `kDifKmacMaxCustomizationStringOverhead`: the only really requirement is that `kDifKmacMaxCustomizationStringOverhead` is at least 3.

Reply:
Done